packages/salesforcedx-vscode-apex/src/index.ts

```diff
@@ -12,23 +12,84 @@ import {
   DEBUGGER_LINE_BREAKPOINTS
 } from './constants';
 import * as languageServer from './languageServer';
+import { ApexTestOutlineProvider } from './views/testOutline';
+
+export type ApexTestRequestInfo = {
+  methodName: string;
+  parent: string;
+  line: number;
+  file: string;
+};
 
 let languageClient: LanguageClient | undefined;
 let languageClientReady = false;
 
 export async function activate(context: vscode.ExtensionContext) {
   languageClient = await languageServer.createLanguageServer(context);
-  const handle = languageClient.start();
-  context.subscriptions.push(handle);
+  if (languageClient) {
+    const handle = languageClient.start();
+    context.subscriptions.push(handle);
+
+    languageClient.onReady().then(() => {
+      languageClientReady = true;
+    });
+  }
+
+  // Test View
+  const rootPath = vscode.workspace.rootPath || '';
+  let apexClasses: ApexTestRequestInfo[] | null = null;
+  if (isLanguageClientReady()) {
+    apexClasses = (await getApexTests()) as ApexTestRequestInfo[];
+  }
+
+  const apexClassesDocs = await vscode.workspace.findFiles('**/*.cls');
```

Comment:
What is this for? Also, this is not correct. You cannot just find all the .cls files in a project. You need to look into sfdx-project.json and only look into parts that are in the packageDirectories. That is what the Apex LSP does.

So, if you do it this way, there is a possibility that the results will be different.

Reply:
I know sometimes when I used debug to run an extension, I would run into problems with using lsp. I am using this as a backup. If that is not necessary I can get rid of this.

Reply:
Could you elaborate more on the issue with the LSP? 
I think it would be nice to not have to use this manual approach.

Reply:
When I first launch the extension, isLanguageClientReady() will  return false (probably has to do with the fact that activate is async and and createLanuageServer is called in activate). When that happens, I use the manual parsing approach.